src/components/Button/index.tsx

```diff
@@ -1,19 +1,26 @@
 /** @jsx jsx */
 import * as React from "react";
-import { jsx, css } from "@emotion/core";
-import t from "../../../lib/theme";
+import { jsx } from "@emotion/core";
+import { Theme } from "../../types/global";
+import Text from "../Text";
 
 interface Props {
   text?: string;
 }
 
-const styles = css({
-  background: t.colors.mojogreen,
-  color: "white",
-});
-
 const Button: React.SFC<Props> = ({ text, children }) => (
-  <button css={styles}>{text || children}</button>
+  <button
+    css={({ ...theme }: Theme) => ({
+      paddingBottom: theme.spacing[2],
+      paddingLeft: theme.spacing[4],
+      paddingRight: theme.spacing[4],
+      paddingTop: theme.spacing[2],
```

Comment:
This seems really nonintuitive to access spacing values based on array index. Wouldn't a regular ol' object be more explicit? 

```ts
spacing: {
  xs: "calc(4px + (8 - 4) * ((100vw - 400px) / (1800 - 600)))",
  s: "calc(8px + (16 - 8) * ((100vw - 400px) / (1800 - 600)))",
  m: "calc(16px + (32 - 16) * ((100vw - 400px) / (1800 - 600)))",
  ml: "calc(44px + (88 - 44) * ((100vw - 400px) / (1800 - 600)))",
  l: "calc(88px + (176 - 88) * ((100vw - 400px) / (1800 - 600)))",
  xl: "calc(176px + (352 - 176) * ((100vw - 400px) / (1800 - 600)))",
}
```
We could alternatively use their long names as well.

Reply:
I think swapping to something like that hinders the flexibility a bit. We may eventually need to add more spacing sizes. Naming would eventually become difficult, and hard to maintain in my opinion.

Reply:
Yeah we decided to run with the array syntax for the added flexibility as chris mentioned, although it _looks_ weirder at first, it think its a bit more clear than potentially having non normal values like `ml`, which would open up the gate to things like `ms`, `mxl`... 

since the design is still fluid, and we want the ui library to maintain maximum flexibility, i think trading away conventional syntax is worth it.